### ISSUE TITLE
[WSL] Styles Not Applying on When Overriding in Page Resources

### ISSUE THREAD

**Issue by Kunal22shah** [unoplatform/uno#18132]
### Current behavior

When overriding button styles in the main page resources using the same key as the global resource, the styles do not apply on WSL (Desktop), but they do on Windows (with latest toolkit : `<UnoToolkitVersion>6.2.0-dev.41</UnoToolkitVersion>` ). A workaround involves adding `BasedOn="{StaticResource DefaultButtonStyle}"` to the button style, which makes it work on WSL as well.
Repro : [UnoApp5.zip](https://github.com/user-attachments/files/16897310/UnoApp5.zip) (no blue background for the buttons)
![RE-bug](https://github.com/user-attachments/assets/d1f3b83a-71dc-4f78-97d8-49a7ade2ff6f)

### Expected behavior

Same blue background for both buttons

### Workaround

adding `BasedOn="{StaticResource DefaultButtonStyle}"` to the button style in the Page Resources

### Works on UWP/WinUI

None

**Comment by agneszitte** [comment]
cc @MartinZikmund, @kazo0 for info

**Comment by agneszitte** [comment]
Related Toolkit issue: https://github.com/unoplatform/uno.toolkit.ui/issues/1187

**Comment by kazo0** [comment]
I don't think this is necessarily constrained to WSL. I can repro on net8.0-desktop windows

**Comment by kazo0** [comment]
Probably related? https://github.com/unoplatform/uno/issues/18115

**Comment by Kunal22shah** [comment]
i tested some cases for this issue with the sample app from above w latest stable sdk :
 
 1. (Page Resources, without BasedOn): Styles did not apply.
 2. (Page Resources, BasedOn included): Styles applied successfully.
 3. (Global Styles from Buttons.xaml removed from page resource) : Styles applied successfully.
 4. (Inline Styles on Button, removed from global and page resource): Styles applied successfully.

**Comment by kazo0** [comment]
Ok, here's what is happening. Both WinUI and Uno have a generic.xaml that contains all of the default implicit styles
 
So if I have these two styles:

```xml
<Style x:Key="ActionButtonStyle"
		   TargetType="Button">
		<Setter Property="utu:ResourceExtensions.Resources">
			<Setter.Value>
				<ResourceDictionary>
					<ResourceDictionary.ThemeDictionaries>
						<ResourceDictionary x:Key="Default">
							<StaticResource x:Key="ButtonBackground" ResourceKey="ActionBackgroundBrush" />
							<StaticResource x:Key="ButtonForeground" ResourceKey="ActionForegroundBrush" />
							<StaticResource x:Key="ButtonBackgroundPointerOver" ResourceKey="Gray300Brush" />
							<StaticResource x:Key="ButtonBackgroundPressed" ResourceKey="ActionBackgroundBrush" />
						</ResourceDictionary>
					</ResourceDictionary.ThemeDictionaries>
				</ResourceDictionary>
			</Setter.Value>
		</Setter>
</Style>

<Style x:Key="ActionButtonStyle2"
		   TargetType="Button"
		   BasedOn="{StaticResource DefaultButtonStyle}">
		<Setter Property="utu:ResourceExtensions.Resources">
			<Setter.Value>
				<ResourceDictionary>
					<ResourceDictionary.ThemeDictionaries>
						<ResourceDictionary x:Key="Default">
							<StaticResource x:Key="ButtonBackground" ResourceKey="ActionBackgroundBrush" />
							<StaticResource x:Key="ButtonForeground" ResourceKey="ActionForegroundBrush" />
							<StaticResource x:Key="ButtonBackgroundPointerOver" ResourceKey="Gray300Brush" />
							<StaticResource x:Key="ButtonBackgroundPressed" ResourceKey="ActionBackgroundBrush" />
						</ResourceDictionary>
					</ResourceDictionary.ThemeDictionaries>
				</ResourceDictionary>
			</Setter.Value>
		</Setter>
</Style>
```

And this as my xaml:

```xml
<StackPanel HorizontalAlignment="Center"
			VerticalAlignment="Center"
			Orientation="Horizontal">

	<Button Content="Menu"
			Style="{StaticResource ActionButtonStyle}" />

	<Button Margin="12,0,0,0"
			Content="Menu 2"
			Style="{StaticResource ActionButtonStyle2}" />

</StackPanel>
```

I see this on Windows:

![image](https://github.com/user-attachments/assets/877ab764-5432-4398-9fcf-e76473e75a3a)

And this on Uno platforms:

![image](https://github.com/user-attachments/assets/d1afeb71-6c2e-4797-8bf2-681b61a1f8ae)


So we can see that the style without BasedOn is different on both platforms. So since there's no BasedOn, it resolves the default style to the one in Generic.xaml.
 
Generic.xaml for WinUI: https://github.com/microsoft/microsoft-ui-xaml/blob/14364e6c301e881c5e2b6b9c39503343c66e53eb/src/dxaml/xcp/dxaml/themes/generic.xaml#L5984
 
Uno's Generic.xaml: https://github.com/unoplatform/uno/blob/4dabe7d629771d24e312a9da948f24241387b167/src/Uno.UI/UI/Xaml/Style/Generic/Generic.xaml#L2898
 
So it is CORRECT that the two buttons in the app are different. BUT the reason it looks like the ResourceExtension isn't properly overriding the resources for the non-BasedOn style is because of the difference between Uno's default generic.xaml Button style and WinUI's generic.xaml Button style.
 
See the one in WinUI is using the ButtonBackground resource (https://github.com/microsoft/microsoft-ui-xaml/blob/14364e6c301e881c5e2b6b9c39503343c66e53eb/src/dxaml/xcp/dxaml/themes/generic.xaml#L5985)
 
But the Uno one is not: https://github.com/unoplatform/uno/blob/4dabe7d629771d24e312a9da948f24241387b167/src/Uno.UI/UI/Xaml/Style/Generic/Generic.xaml#L2900-L2901

So it is normal that the BasedOn is required if you are indeed looking to override the Fluent "DefaultButtonStyle"
 
What should be fixed is we should be updating the Generic.xaml in Uno to match the styles in WinUI's Generic.xaml

**Comment by MartinZikmund** [comment]
Closing in favor of https://github.com/unoplatform/uno/issues/12839
